### PR TITLE
Collapsible World Map

### PR DESCRIPTION
**Pull request by chevreaux**
Added a collapsible element and made the world map collapsible, so we have more space to edit the level.

(dangnamit i'm still making commits with my other account, i thought i fixed that)

### CODE REVIEW

**Comment by ngyikp** [comment]
I just added CI in a recent commit, please take a look at the lint failures, you should be able to easily fix them with `npm run lint -- --fix` and `npm run prettier -- --write`

If you use VSCode, you can install the ESLint, Flow Language Support and Prettier extensions so you get nice lint feedback inside the editor.

I did thought of putting a collapse map button, but wouldn't it mean additional height is taken up just to show the collapse/expander button? :P

File: src/Collapsible.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+type Props = $ReadOnly<{
+	children: React$Node,
+    buttonText: String,
+}>;
+
+export default function Collapsible(props: Props): React$Node {
+    const [isOpen, setIsOpen] = useState(true)
+    const styleOpen = {height: 'auto', overflow: 'initial'}
+    const styleClosed = {height: '0px', overflow: 'hidden'};
```
Comment (ngyikp) [inline src/Collapsible.js:12]: Can be moved to the CSS file

**Comment by chevreaux** [comment]
Haha, I've only just now realized the world map is resizable. This isn't the most important feature then. Though I think it's still worth adding.

**Comment by ngyikp** [comment]
I think it makes more sense to build a new settings screen as there could be more settings such as hiding the sidebar map to save screen space

I added react-modal for the new terrain editor so I could use that as a starting point

**Comment by chevreaux** [comment]
Gonna fix these check failures to make sure I've got the process right. Up to you if wanna merge or not.

**Comment by chevreaux** [comment]
ok last try, if this still doesn't work I will cry

**Comment by chevreaux** [comment]
Theeere we go. Flow is an asshole. Feel free to close or merge.

**Pull request closed without merging**